readme.md

```diff
@@ -23,6 +23,155 @@ var Component = require('react-view-models').Component;
+import DefineMap from 'can-define/map/';
+
+export default class AppComponent extends Component {
+  render() {
+    return <div>{this.props.text}</div>;
+  }
+}
+
+AppComponent.ViewModel = DefineMap.extend('AppVM', {
+  first: {
+    type: 'string',
+    value: 'foo'
+  },
+  last: {
+    type: 'string',
+    value: 'bar'
+  },
+  text: {
+    get() {
+      return this.first + this.last;
+    },
+  },
+});
+```
+
+### `makeReactComponent( displayName, CanComponent )`
+Convert a CanComponent class into a React Component.
+
+`makeReactComponent()` takes 2 arguments. The first (optional) is the displayName of the ReactComponent. The second argument is a CanComponent constructor function. The `makeReactComponent()` function returns a React Component which can then be imported and used in any react component or render function as usual.
+
+Since the Component doesn't produce DOM artifacts of it’s own, you won’t end up with any wrapper divs or anything to worry about, but in react-device-tools you will see the component with the `displayName` (or defaults to `CanComponentWrapper`) in the tree.
+
+#### Example
+
+```javascript
+import React from 'react';
+import CanComponent from 'can-component';
+import { makeReactComponent } from 'react-view-models';
+
+const InnerComponent = makeReactComponent(
+  CanComponent.extend('InnerComponent', {
+    tag: 'inner-component',
+    view: stache('<div class="inner">{{text}}</div>')
+  })
+);
+
+export default class AppComponent extends Component {
+  render() {
+    return (
+      <InnerComponent text="inner text" />
+    );
+  }
+}
+```
+
+## Common use cases when using a view model
+Here are some examples that may come up when using a view-model that may not be obvious at first:
+
+### Transforming a prop before passing it down to a child component
+
+Sometimes you want a prop that is set on your connected component to be set to the exact same prop key on the child component, but modified slightly before passing it down. Here's an example of that:
+
+```javascript
+const ViewModel = DefineMap.extend({
+  someProp: {
+    set( newVal ) {
+      return newVal.toUpperCase();
+    }
+  }
+});
+```
+
+### Calling a parents callback, while also doing something special in your view models callback
+Sometimes, you still want to notify the connected components owner component that the state changed (by calling a callback), but only after or while doing something different within the view-model. In this case, you'll want to define the callback prop as a observable attribute with a getter, rather than a method, and use the `lastSetVal` argument to call the parent components callback.
+
+```javascript
+const ViewModel = DefineMap.extend({
+  onChange: {
+    type: 'function',
+    get( lastSetValue ) {
+      return (ev) => {
+        this.changeTheThing(ev.target);
+        if ( lastSetValue ) {
+          return lastSetValue(ev);
+        }
+      };
+    }
+  }
+});
+```
```

Comment:
Maintaining the readme and the docs on the site is a burden. Instead, I would include the bare minimum to get someone interested and then point them to the full docs on the site.

[I realize we have generated docs in a lot of our modules, and I’m working towards moving us to just have the bare minimum in the readme and then get people onto our site to read the full docs.]